snowpack/src/util.ts

```diff
@@ -325,3 +325,39 @@ export function cssSourceMappingURL(code: string, sourceMappingURL: string) {
 export function jsSourceMappingURL(code: string, sourceMappingURL: string) {
   return code.replace(/\n*$/, '') + `\n//# sourceMappingURL=${sourceMappingURL}\n`; // strip ending lines & append source map (with linebreaks for safety)
 }
+
+/** URL relative */
+export function relativeURL(path1: string, path2: string): string {
+  let url = path.relative(path1, path2).replace(/\\/g, '/');
+  if (
+    !url.startsWith('.') &&
+    !url.startsWith('/') &&
+    !url.startsWith('http://') &&
+    !url.startsWith('https://')
+  ) {
+    url = './' + url;
+  }
+  return url;
+}
+
+const CLOSING_HEAD_TAG = /<\s*\/\s*head\s*>/gi;
+
+/** Append HTML before closing </head> tag */
+export function appendHTMLToHead(doc: string, htmlToAdd: string) {
+  const closingHeadMatch = doc.match(CLOSING_HEAD_TAG);
+  if (closingHeadMatch && closingHeadMatch.length === 1) {
+    return doc.replace(new RegExp(`(${closingHeadMatch[0]})`), `${htmlToAdd}$1`);
+  }
+  return doc + htmlToAdd;
+}
+
+const CLOSING_BODY_TAG = /<\s*\/\s*body\s*>/gi;
+
+/** Append HTML before closing </body> tag */
+export function appendHTMLToBody(doc: string, htmlToAdd: string) {
+  const closingBodyMatch = doc.match(CLOSING_BODY_TAG);
+  if (closingBodyMatch && closingBodyMatch.length === 1) {
+    return doc.replace(new RegExp(`(${closingBodyMatch[0]})`), `${htmlToAdd}$1`);
+  }
+  return doc + htmlToAdd;
+}
```

Comment:
Since this is a bit more essential to get right (it's for the final build that goes out to users on all sorts of browsers, and not your local development on your modern browser) I'd reccomend we error out loadly if we find more than one </head> or </body> tag, instead of silently falling back. 

I'm actually pretty blown away that there's no good npm package for this, and that the only well-used ones are written for gulp and webpack. Parse5 could be a good alternative as well, although I'm fine with starting with our regex and then adding something more complex if we hear about a lot of users with commented out head and body tags :) 

https://www.npmjs.com/package/parse5

Reply:
Another few options I found that seem light.

https://github.com/mykolaharmash/hyntax
https://github.com/fb55/htmlparser2
https://github.com/henrikjoreteg/html-parse-stringify

Reply:
Dug into this a little more, and I think the problem is it’s easy to find a _parser_, but hard to mutate easily (e.g. insert nodes) with that same library. I did find some libraries that can do this, but it turns out to be a pretty complex thing in Node-land. Will look into it a little further.

Reply:
Good news! `htmlparser2` + their `domutils` helper is actually a decent solution to this. But since in other areas we’re scanning for `<script>` tags in HTML, too, WDYT about exploring this in a followup PR?

This RegEx is pretty safe because we guarantee we’re not rewriting any HTML; whenever you introduce a parser, mutate something, and restringify, there’s always a chance the parser dropped something or reformatted the document (I only have this fear after playing around with a few and seeing our tests fail because stuff went missing from the HTML like comments).

Anyway, if we explore this in a followup PR, we could possibly clean up some other RegExs dealing with HTML outside the scope of this PR. The impact of a change would blow out this scope/focus quite a bit (and this PR adds a test, anyway, so we’d catch a regression to this new plugin).

Reply:
yea, I'm +1. The more I think about it, the more rare I even think a "two </body> tags" problem even is. Definitely feels like we're getting into YAGNI territory here :)